src/core/trulens/core/database/sqlalchemy.py

```diff
@@ -715,6 +717,129 @@ def get_records_and_feedback(
 
             return AppsExtractor().get_df_and_cols(records=records)
 
+    def insert_ground_truth(
+        self, ground_truth: mod_groundtruth_schema.GroundTruth
+    ) -> mod_types_schema.GroundTruthID:
+        """See [DB.insert_ground_truth][trulens.core.database.base.DB.insert_ground_truth]."""
+
+        # TODO: thread safety
+        with self.session.begin() as session:
+            if (
+                _ground_truth := session.query(self.orm.GroundTruth)
+                .filter_by(ground_truth_id=ground_truth.ground_truth_id)
+                .first()
+            ):
+                _ground_truth.ground_truth_json = ground_truth.model_dump_json()
+            else:
+                _ground_truth = self.orm.GroundTruth.parse(
+                    ground_truth, redact_keys=self.redact_keys
+                )
+
+                session.merge(_ground_truth)
+
+            logger.info(
+                f"{UNICODE_CHECK} added ground truth {_ground_truth.ground_truth_id}"
+            )
+
+            return _ground_truth.ground_truth_id
+
+    def batch_insert_ground_truth(
+        self, ground_truths: List[mod_groundtruth_schema.GroundTruth]
+    ) -> List[mod_types_schema.GroundTruthID]:
+        """See [DB.batch_insert_ground_truth][trulens_eval.database.base.DB.batch_insert_ground_truth]."""
+        with self.session.begin() as session:
+            ground_truths_list = []
+
+            for ground_truth in ground_truths:
+                if (
+                    _ground_truth := session.query(self.orm.GroundTruth)
```

Comment:
This would be very inefficient.. For every batch call, we would make n queries. And for 80% of the cases, those would be not useful.

Reply:
I would say the current batch API need not support updates and only support inserts

Reply:
I do think idempotency / upsertion support is important. 

I optimized the method to do ```
SELECT *
FROM ground_truth
WHERE ground_truth_id IN (id1, id2, id3, ..., idN);
```

so should be a lot better